api/models/User.js

```diff
@@ -31,4 +40,165 @@ userSchema.statics.create = function(name) {
+			if(err) reject(err);
+			else resolve(user);
+		});
+	});
+};
+
+/**
+ * Delete a User object from the database
+ * @param {string} user's googleID (as hash)
+ * @returns {void} nothing
+ * @example
+ * User.delete('someGoogleIdHash')
+ *     .catch(error => console.error(error));
+ */
+
+userSchema.statics.delete = function(googleId)
+{
+	return new Promise((resolve, reject)=>{
+		User.findOne({googleID: googleId}).remove((err,offer)=>{
+			if(err) reject(err);
+			resolve(offer);
+		});
+	});
+};
+
+/**
+ * Change User's isAdmin status
+ * @param {Boolean} User's new isAdmin status
+ * @returns {User} changed user object
+ * @example
+ * User.read('someGoogleIdHash')
+ *     .then(user => {
+ *	        user.setAdminStatus(true);
+ *     })
+ *	   .catch(error => console.error(error));
+ */
+
+
+userSchema.methods.setAdminStatus = function(adminStatus)
+{
+	this.isAdmin = adminStatus;
+	
+	return new Promise((resolve, reject)=>{
+		this.save((error, user) => {
+			if(error) reject(error);
+			resolve(user);
+		});
+	});
+};
+
+/**
+ * Change User's isMentor status
+ * @param {Boolean} User's new isMentor status
+ * @returns {User} changed user object
+ * @example
+ * User.read('someGoogleIdHash')
+ *     .then(user => {
+ *	        user.setMentorStatus(true);
+ *     })
+ *	   .catch(error => console.error(error));
+ */
+
+userSchema.methods.setMentorStatus = function(mentorStatus)
+{
+	this.isMentor = mentorStatus;
+
+	return new Promise((resolve,reject)=>{
+		this.save((err, user)=>{
+			if(err) reject(err);
+			else resolve(user);
+		});
+	});
+};
+
+
+/**
+ * Add a new skill to the User object
+ * @param {string} User's new skill
+ * @returns {User} changed user object
+ * @example
+ * User.read('someGoogleIdHash')
+ *     .then(user => {
+ *	        user.addSkill('python');
+ *     })
+ *	   .catch(error => console.error(error));
+ *
+ * If attempt to add a skill that already exists, 
+ * an error will be thrown
+ */
+
+userSchema.methods.addSkill = function(skill)
+{
+	//skill does not exist if index is -1;
+	//skill exists already if index is >=0;
+	const index = this.skills.indexOf(skill);
+
+	if(index >=0 ){
+		return new Promise(new Error('skill: '+skill+' already exists !'));
+	}
+
+	this.skills.push(skill);
+
+	return new Promise((resolve, reject)=>{
+		this.save((err,user)=>{
+			if(err) reject(err);
+			else resolve(user);
+		});
+	});
+};
+
+/**
+ * Remove a skill from the User object
+ * @param {string} skill to remove from the User object
+ * @returns {User} changed user object
+ * @example
+ * User.read('someGoogleIdHash')
+ *     .then(user => {
+ *	        user.removeSkill('python');
+ *     })
+ *	   .catch(error => console.error(error));
+ *
+ * If attempt to remove a skill that does not exists, 
+ * an error will be thrown
+ */
+
+userSchema.methods.removeSkill = function(skill)
+{
+	const index = this.skills.indexOf(skill);
+
+	if(index < 0 ){
+		return new Promise(new Error('skill: '+skill+' does not exists !'));
```

Comment:
I believe you don't put spaces before exclamation marks in English. (I.e., it's not French.)